components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -66,6 +82,30 @@ interface PushConnection : Closeable {
     fun isInitialized(): Boolean
 }
 
+/**
+ * An implementation of [PushConnection] for the native component using the [PushAPI].
+ *
+ * Implementation notes: There are few important concepts to note here - for WebPush, the identifier that is required
+ * for notifying clients is called a "scope". This can be a site's host URL or an other uniquely identifying string to
+ * create a push subscription for that given scope.
+ *
+ * In the native Rust component the concept is similar, in that we have a unique ID for each subscription, however
+ * the ID is called a "channel ID" (or chid), which is a UUID. In the implementation, the scope can also be provided
+ * to the native layer, however it is mostly just stored in the database but not used in any significant way.
```

Comment:
Maybe worth calling out that it _may_ be used for stuff in the future, e.g. rate limiting, but without immediate plans to do so (e.g. no tracking issues in github!)